ros_ign_gazebo/include/ros_ign_gazebo/Stopwatch.hpp

```diff
@@ -0,0 +1,182 @@
+// Copyright 2022 Open Source Robotics Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROS_IGN_GAZEBO__STOPWATCH_HPP_
+#define ROS_IGN_GAZEBO__STOPWATCH_HPP_
+
+#include <rclcpp/clock.hpp>
+
+#include <chrono>
+#include <memory>
+
+
+namespace ros_ign_gazebo
+{
+// Forward declarations.
+class StopwatchPrivate;
+
+/// \brief The stopwatch accepts an rclcpp::Clock instance from a rclcpp::Node (allowing use of
+/// sim time if use_sim_time set in node). Keeps track of time spent in the run state, accessed
+/// through ElapsedRunTime(), and time spent in the stop state, accessed through ElapsedStopTime().
+/// Elapsed run time starts accumulating after the first call to Start(). Elapsed stop time starts
+/// accumulation after Start() has been called followed by Stop(). The stopwatch can be reset with
+/// the Reset() function.
+///
+/// # Example usage
+///
+/// ```{.cpp}
+/// ros_ign_gazebo::Stopwatch watch;
+/// watch.Start();
+///
+/// // do something...
+///
+/// std::cout << "Elapsed time is "
+/// << std::chrono::duration_cast<std::chrono::milliseconds>(
+///   timeSys.ElapsedRunTime()).count() << " ms\n";
+/// watch.Stop();
+/// ```
+class Stopwatch
+{
+  /// \brief Constructor.
+
+public:
+  Stopwatch();
+
+  /// \brief Copy constructor
+  /// \param[in] _watch The stop watch to copy.
+
+public:
+  Stopwatch(const Stopwatch & _watch);
```

Comment:
Please use this style
```suggestion
  /// \brief Copy constructor
  /// \param[in] _watch The stop watch to copy.
  public: Stopwatch(const Stopwatch & _watch);
```

Reply:
Uncrustify is failing with that style though